src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/ObjectExplorerService.cs

```diff
@@ -346,8 +346,19 @@ public ObjectExplorerSession(string uri, TreeNode root, IMultiServiceProvider se
 
             public static ObjectExplorerSession CreateSession(ConnectionCompleteParams response, IMultiServiceProvider serviceProvider)
             {
-                TreeNode serverNode = new ServerNode(response, serviceProvider);
-                return new ObjectExplorerSession(response.OwnerUri, serverNode, serviceProvider, serviceProvider.GetService<ConnectionService>());
+                TreeNode rootNode = new ServerNode(response, serviceProvider);
+                var session = new ObjectExplorerSession(response.OwnerUri, rootNode, serviceProvider, serviceProvider.GetService<ConnectionService>());
+                if (!ObjectExplorerUtils.IsSystemDatabaseConnection(response.ConnectionSummary.DatabaseName))
+                {
+                    var children = rootNode.Expand();
+                    var databasesRoot = children.FirstOrDefault(x => x.NodeTypeId == NodeTypes.Databases);

```

Comment:
Will write the comment on assumption. If we change the structure, it will break the OE tests

---
In reply to: [112340512](https://github.com/Microsoft/sqltoolsservice/pull/322#discussion_r112340512) [](ancestors = 112340512)